treeherder/log_parser/intermittents.py

```diff
@@ -0,0 +1,134 @@
+from treeherder.model.models import Group, GroupStatus, Job, Push
+
+
+def check_and_mark_intermittent(job_id):
+    current_job = Job.objects.get(id=job_id)
+    jtname = current_job.job_type.name.strip("-cf")
+    ids = [current_job.push.id]
+
+    try:
+        _ = int(jtname.split("-")[-1])
+        jtname = "-".join(jtname.split("-")[:-1])
+    except ValueError:
+        pass
+
+    # if we are not on try, look at recent history
+    if current_job.repository.id != 4:
+        # get list of pushes
+        ids = Push.objects.filter(repository__id=current_job.repository.id).values("id")[:20]
```

Comment:
Should there be conditions which 20 pushes are requested?

Reply:
I think this is an error, I need to start at current pushid and go back 20, possibly this can be an easy math equation.

Reply:
it appears push.id is sequential when including all branches, so current.push.id-20 could be on a different branch.  I have fixed this logic in my recent commit.